### PR TITLE
Add guide for RHEL 8 #3800 

### PR DESCRIPTION
**Pull request by Ganzabahl**
#### Summary

Added RHEL 8 installation documentation. Have tested by building locally and I'm happy with how it looks similar to the other pages.

#### Ticket Link

  Fixes https://github.com/mattermost/docs/issues/3800

### CODE REVIEW

**Comment by justinegeffen** [comment]
Thanks, @Ganzabahl! Note: I recommend not merging this until https://github.com/mattermost/docs/pull/3970 is merged, as there are substantial changes which may affect this PR. I've added the ``Do Not Merge`` label for this reason.

**Comment by Ganzabahl** [comment]
Hi @justinegeffen . 
I'm a bit confused if this means my contribution will have to be redone or rewritten?
Depending on how fast changes gets merged in, you could merge #3970 and let me have a look if we are safe to go or minor changes needs to be done.
In case #3970 doesn't get merged anytime soon, how does it affect the Hacktober progress? 
Sorry I'm asking because I'm happy to help but this is my first Hacktober and really hyped about it. Was unaware about being able to contribute etc before a friend told me about Hacktober :)

**Comment by justinegeffen** [comment]
Thanks @Ganzabahl. Apologies if I was unclear. Your PR will definitely not have to be rewritten. :) I am reviewing https://github.com/mattermost/docs/pull/3970 and will sync up with @svelle later today for his review. @hanzei, I'd like to just confirm that @Ganzabahl's Hacktoberfest progress isn't hampered by PRs awaiting review? :)

File: source/install/config-ssl-http2-nginx.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ You can use any certificate that you want, but these instructions show you how t
 
   ``sudo service nginx stop``
 
-  On Ubuntu 16.04, Ubuntu 18.04 and RHEL 7:
+  On Ubuntu 16.04, Ubuntu 18.04, RHEL 7 and RHEL 8:
```
Comment (justinegeffen) [inline source/install/config-ssl-http2-nginx.rst:42]:
```suggestion
  On Ubuntu 18.04, RHEL 7, and RHEL 8:
```

File: source/install/config-ssl-http2-nginx.rst [hunk 2]
```diff
@@ -108,7 +108,7 @@ You can use any certificate that you want, but these instructions show you how t
 
   ``sudo service nginx start``
 
-  On Ubuntu 16.04, Ubuntu 18.04 and RHEL 7:
+  On Ubuntu 16.04, Ubuntu 18.04, RHEL 7 and RHEL 8:
```
Comment (justinegeffen) [inline source/install/config-ssl-http2-nginx.rst:111]:
```suggestion
  On Ubuntu 18.04, RHEL 7, and RHEL 8:
```

File: source/install/install-rhel-8-mattermost.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
+..  _install-rhel-8-mattermost:
+
+Installing Mattermost Server
+============================
+
+Install Mattermost Server on a 64-bit machine.
+
+Assume that the IP address of this server is 10.10.10.2
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:8]:
```suggestion
Assume that the IP address of this server is ``10.10.10.2``.
```

File: source/install/install-rhel-8-mattermost.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  .. note::
+    The storage directory will contain all the files and images that your users post to Mattermost, so you need to make sure that the drive is large enough to hold the anticipated number of uploaded files and images.
+
+6. Set up a system user and group called ``mattermost`` that will run this service, and set the ownership and permissions.
+
+  a. ``sudo useradd --system --user-group mattermost``
+  b. ``sudo chown -R mattermost:mattermost /opt/mattermost``
+  c. ``sudo chmod -R g+w /opt/mattermost``
+
+7. Set up the database driver in the file ``/opt/mattermost/config/config.json``. Open the file as root in a text editor and make the following changes:
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:39]:
```suggestion
7. Set up the database driver in the file ``/opt/mattermost/config/config.json``. Open the file as *root* in a text editor and make the following changes:
```

File: source/install/install-rhel-8-mattermost.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+    a. Change to the ``mattermost`` directory:
+      ``cd /opt/mattermost``
+    b. Start the Mattermost server as the user mattermost:
+
+      ``sudo -u mattermost ./bin/mattermost``
+
+  When the server starts, it shows some log information and the text ``Server is listening on :8065``. You can stop the server by pressing CTRL+C in the terminal window.
+
+10. Set up Mattermost to use the systemd init daemon which handles
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:62]:
```suggestion
10. Set up Mattermost to use the ``systemd init`` daemon which handles
```

File: source/install/install-rhel-8-mattermost.rst [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+   supervision of the Mattermost process.
+
+  a. Create the Mattermost configuration file:
+
+    ``sudo touch /etc/systemd/system/mattermost.service``
+
+  b. Open the configuration file in your favorite text editor, and copy the following lines into the file:
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:69]:
```suggestion
  b. Open the configuration file in a text editor, and copy the following lines into the file:
```

File: source/install/install-rhel-8-mattermost.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    a. Change to the ``mattermost`` directory:
+      ``cd /opt/mattermost``
+    b. Start the Mattermost server as the user mattermost:
+
+      ``sudo -u mattermost ./bin/mattermost``
+
+  When the server starts, it shows some log information and the text ``Server is listening on :8065``. You can stop the server by pressing CTRL+C in the terminal window.
+
+10. Set up Mattermost to use the systemd init daemon which handles
+   supervision of the Mattermost process.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:63]: This line should be added to the previous one (i.e., a single line sentence without a line break). :)

File: source/install/install-rhel-8-mattermost.rst [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+      ExecStart=/opt/mattermost/bin/mattermost
+      PIDFile=/var/spool/mattermost/pid/master.pid
+      TimeoutStartSec=3600
+      LimitNOFILE=49152
+
+      [Install]
+      WantedBy=multi-user.target
+
+    .. note::
+      If you are using MySQL, replace ``postgresql.service`` by ``mysqld.service`` in the ``[unit]`` section.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mattermost.rst:90]:
```suggestion
      If you are using MySQL, replace ``postgresql.service`` with ``mysqld.service`` in the ``[unit]`` section.
```

File: source/install/install-rhel-8-mysql.rst [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  
+  .. note::
+    1. The first time that you start MySQL, the superuser account ``'root'@'localhost'`` is created without a password.
+    2. Also the first time that you start MySQL, the ``validate_password`` plugin is installed. The plugin forces passwords to contain at least one upper case letter, one lower case letter, one digit, and one special character, and that the total password length is at least 8 characters.
+
+4. Change the root password. Login with the the command below. Since the password is blank by default just press ENTER when prompted for a password.
+
+  ``mysql -u root -p``
+
+5. Change the password. At the mysql prompt, type the following command. Be sure to replace ``Password42!`` with the password that you want to use.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mysql.rst:28]:
```suggestion
5. Change the password. At the prompt, type the following command. Be sure to replace ``Password42!`` with the password that you want to use:
```

File: source/install/install-rhel-8-postgresql.rst [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+11. Exit the PostgreSQL interactive terminal.
+
+  ``postgres=# \q``
+
+12. Log out of the *postgres* account.
+
+  ``exit``
+
+13. (Optional) If you use a different server for your database and the Mattermost app server, you may allow PostgreSQL to listen on all assigned IP Addresses. To do so, open ``/etc/postgresql/main/postgresql.conf`` as root in a text editor. As a best practice, ensure that only the Mattermost server is able to connect to the PostgreSQL port using a firewall.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:56]:
```suggestion
13. (Optional) If you use a different server for your database and the Mattermost app server, you may allow PostgreSQL to listen on all assigned IP Addresses. To do so, open ``/etc/postgresql/main/postgresql.conf`` as *root* in a text editor. As a best practice, ensure that only the Mattermost server is able to connect to the PostgreSQL port using a firewall.
```

File: source/install/install-rhel-8-postgresql.rst [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+    ``listen_addresses = '*'``
+
+  d. Restart PostgreSQL for the change to take effect:
+
+    ``sudo systemctl restart postgresql``
+
+14. Modify the file ``pg_hba.conf`` to allow the Mattermost server to communicate with the database.
+
+  **If the Mattermost server and the database are on the same machine**:
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:74]:
```suggestion
  **If the Mattermost server and the database are on the same machine:**
```

File: source/install/install-rhel-8-postgresql.rst [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+    b. Find the following line:
+
+      ``local   all             all                        peer``
+
+    c. Change ``peer`` to ``trust``:
+
+      ``local   all             all                        trust``
+
+  **If the Mattermost server and the database are on different machines**:
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:86]:
```suggestion
  **If the Mattermost server and the database are on different machines:**
```

File: source/install/install-rhel-nginx.rst [hunk 13]
```diff
@@ -12,13 +12,15 @@ The main benefits of using a proxy are as follows:
   -  Port mapping ``:80`` to ``:8065``
   -  Standard request logs
 
-**To install NGINX on RHEL 6 or 7:**
+**To install NGINX on RHEL 6, 7 or 8:**
```
Comment (justinegeffen) [inline source/install/install-rhel-nginx.rst:15]:
```suggestion
**To install NGINX on RHEL 7 or 8:**
```

File: source/install/install-rhel-nginx.rst [hunk 14]
```diff
@@ -12,13 +12,15 @@ The main benefits of using a proxy are as follows:
...
 
 1. Log in to the server that will host the proxy, and open a terminal window.
 
 2. Create the file /etc/yum.repos.d/nginx.repo.
   ``sudo touch /etc/yum.repos.d/nginx.repo``
 
+  If you are on RHEL 8 you can skip to step 4. Install NGINX.
```
Comment (justinegeffen) [inline source/install/install-rhel-nginx.rst:22]:
```suggestion
  If you are on RHEL 8 you can skip to **Step 4. Install NGINX**.
```

File: source/install/install-rhel-nginx.rst [hunk 15]
```diff
@@ -12,13 +12,15 @@ The main benefits of using a proxy are as follows:
...
+
 3. Open the file as root in a text editor and add the following contents, where *{version}* is **6** for RHEL 6, and **7** for RHEL 7:
```
Comment (justinegeffen) [inline source/install/install-rhel-nginx.rst:24]:
```suggestion
3. Open the file as *root* in a text editor and add the following contents, where *{version}* is **7** for RHEL 7:
```

File: source/install/prod-rhel-8.rst [hunk 16]
```diff
@@ -0,0 +1,6 @@
+..  _prod-rhel-8:
+
+Production Install on RHEL 8+
+===============================
+
+This document is superseded by :doc:`install-rhel-8`.
```
Comment (justinegeffen) [inline source/install/prod-rhel-8.rst:6]:
```suggestion
This document is replaced by :doc:`install-rhel-8`.
```

File: source/install/install-rhel-8-mysql.rst [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+  ``mysql> ALTER USER 'root'@'localhost' IDENTIFIED BY 'Password42!';``
+
+6. Set MySQL to start automatically when the machine starts.
+
+  ``sudo systemctl enable mysqld``
+
+7. Create the Mattermost user 'mmuser'.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mysql.rst:36]:
```suggestion
7. Create the Mattermost user *mmuser*.
```

File: source/install/install-rhel-8-mysql.rst [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    1. Use a password that is more secure than 'mmuser-password'.
+    2. The '%' means that mmuser can connect from any machine on the network. However, it's more secure to use the IP address of the machine that hosts Mattermost. For example, if you install Mattermost on the machine with IP address 10.10.10.2, then use the following command:
+
+      ``mysql> create user 'mmuser'@'10.10.10.2' identified by 'mmuser-password';``
+
+8. Create the Mattermost database.
+
+  ``mysql> create database mattermost;``
+
+9. Grant access privileges to the user 'mmuser'.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-mysql.rst:50]:
```suggestion
9. Grant access privileges to the user *mmuser*.
```

File: source/install/install-rhel-8-postgresql.rst [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+7. Start the PostgreSQL interactive terminal.
+
+  ``psql``
+
+8.  Create the Mattermost database.
+
+  ``postgres=# CREATE DATABASE mattermost;``
+
+9.  Create the Mattermost user 'mmuser'.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:37]:
```suggestion
9.  Create the Mattermost user *mmuser*.
```

File: source/install/install-rhel-8-postgresql.rst [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+  d. Restart PostgreSQL for the change to take effect:
+
+    ``sudo systemctl restart postgresql``
+
+14. Modify the file ``pg_hba.conf`` to allow the Mattermost server to communicate with the database.
+
+  **If the Mattermost server and the database are on the same machine**:
+
+    a. Open ``/var/lib/pgsql/data/pg_hba.conf`` as root in a text editor.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:76]:
```suggestion
    a. Open ``/var/lib/pgsql/data/pg_hba.conf`` as *root* in a text editor.
```

File: source/install/install-rhel-8-postgresql.rst [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+      ``local   all             all                        peer``
+
+    c. Change ``peer`` to ``trust``:
+
+      ``local   all             all                        trust``
+
+  **If the Mattermost server and the database are on different machines**:
+
+    a. Open ``/var/lib/pgsql/data/pg_hba.conf`` as root in a text editor.
```
Comment (justinegeffen) [inline source/install/install-rhel-8-postgresql.rst:88]:
```suggestion
    a. Open ``/var/lib/pgsql/data/pg_hba.conf`` as *root* in a text editor.
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Some suggestions around formatting but otherwise it looks great, thank you!

Just FYI, some of the suggestions (such as italicizing usernames) are general formatting suggestions as the legacy content doesn't have that formatting - not a reflection on your PR. :)

**Comment by Ganzabahl** [comment]
Your feedback is highly appreciated and I agree on the changes :) 
As you mentioned most of the changes are some overall better practices that has been introduced since the RHEL 7 instructions. So I was happily unaware ;)

**Review by svelle — APPROVED** [review]
lgtm

**Review by wiersgallak — APPROVED** [review]

**Pull request merged**